api/src/main/java/org/apache/iceberg/metrics/DefaultMetricsContext.java

```diff
@@ -24,6 +24,9 @@
 public class DefaultMetricsContext implements MetricsContext {
   private static final int DEFAULT_HISTOGRAM_RESERVOIR_SIZE = 10_000;
 
+  /**
+   * @deprecated will be removed in 2.0.0, use {@link org.apache.iceberg.metrics.Counter} instead.
```

Comment:
Isn't 2.0.0 too far? Should we mention it as "will be removed after 1.0.0"?

Let's wait for @danielcweeks's response on this before addressing this.

Reply:
2.0.0 is correct since we guarantee that we won't break these APIs in minor releases.

Reply:
Thanks for the clarification.

Reply:
In theory you are only allowed to break the API in a major release bump, hence I've 2.0.0 here